backend/arm64_ast.mli

```diff
@@ -214,11 +285,45 @@ module Instruction_name : sig
     | ADDV
 end
 
-module Operand : sig
-  type t
-end
-
 module DSL : sig
+  val reg_op : Reg.t -> Operand.t
```

Comment:
The use of `Reg.t` is super-confusing here, because of `reg.mli` . We should rename `Arm64_ast.Reg` or at least have a CR about it.